biodatasets/n2c2_2018_track1/n2c2_2018_track1.py

```diff
@@ -0,0 +1,296 @@
+    d. Retinopathy
+    e. nephropathy
+    f. neuropathy
+7. ADVANCED-CAD: Advanced cardiovascular disease (CAD).
+For the purposes of this annotation, we define “advanced” as having 2 or more of the following:
+    a. Taking 2 or more medications to treat CAD
+    b. History of myocardial infarction (MI)
+    c. Currently experiencing angina
+    d. Ischemia, past or present
+8. MI-6MOS: MI in the past 6 months
+9. KETO-1YR: Diagnosis of ketoacidosis in the past year
+10. DIETSUPP-2MOS: Taken a dietary supplement (excluding vitamin D) in the past 2 months
+11. ASP-FOR-MI: Use of aspirin to prevent MI
+12. HBA1C: Any hemoglobin A1c (HbA1c) value between 6.5% and 9.5%
+13. CREATININE: Serum creatinine > upper limit of normal
+
+The training consists of 202 patient records with document-level annotations, 10 records
+with textual spans indicating annotator’s evidence for their annotations while test set contains 86.
+
+Note: The inter-annotator average agreement is 84.9%
+"""
+
+_HOMEPAGE = "https://portal.dbmi.hms.harvard.edu/projects/n2c2-nlp/"
+
+_LICENSE = "External Data User Agreement"
+
+_SUPPORTED_TASKS = [Tasks.TEXT_CLASSIFICATION]
+
+_SOURCE_VERSION = "1.0.0"
+
+_BIGBIO_VERSION = "1.0.0"
+
+# Constants
+SOURCE = "source"
+BIGBIO_TEXT = "bigbio_text"
+
+
+def _read_zip(file_path, samples=None):
+    if samples is None:
+        samples = defaultdict(dict)
+    with zipfile.ZipFile(file_path) as zf:
+        for info in zf.infolist():
+
+            base, filename = os.path.split(info.filename)
+            _, ext = os.path.splitext(filename)
+            ext = ext[1:]  # get rid of dot
+            sample_id = filename.split(".")[0]
+
+            if ext == "xml" and not filename.startswith("."):
+                content = zf.read(info).decode("utf-8").encode()
+                root = etree.XML(content)
+                text, tags = root.getchildren()
+                samples[sample_id]["txt"] = text.text
+                samples[sample_id]["tags"] = {}
+                for child in tags:
+                    samples[sample_id]["tags"][child.tag] = child.get("met")
+
+    return samples
+
+
+class N2C22018CohortSelectionDataset(datasets.GeneratorBasedBuilder):
+    """i2b2 2018 cohort selection task"""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    # You will be able to load the "source" or "bigbio" configurations with
+    # ds_source = datasets.load_dataset('my_dataset', name='source')
+    # ds_bigbio = datasets.load_dataset('my_dataset', name='bigbio')
+
+    # For local datasets you can make use of the `data_dir` and `data_files` kwargs
+    # https://huggingface.co/docs/datasets/add_dataset.html#downloading-data-files-and-organizing-splits
+    # ds_source = datasets.load_dataset('my_dataset', name='source', data_dir="/path/to/data/files")
+    # ds_bigbio = datasets.load_dataset('my_dataset', name='bigbio', data_dir="/path/to/data/files")
```

Comment:
same comments about comment removal as in #469 https://github.com/bigscience-workshop/biomedical/pull/469/files#r854751488

Reply:
I assumed these would be of help to the end user, removed in recent commit. :)